lib/xrp/XrpRpc.js

```diff
@@ -0,0 +1,248 @@
+const RippleAPI = require('ripple-lib').RippleAPI;
+const promptly = require('promptly');
+const util = require('util');
+const EventEmitter = require('events');
+
+const passwordPromptAsync = util.promisify(promptly.password);
+
+class XrpRpc {
+  constructor(config) {
+    this.config = config;
+    const {
+      rpcPort,
+      host,
+      protocol,
+      address
+    } = config;
+    const connectionString = `${protocol}://${host}:${rpcPort}`;
+    this.rpc = new RippleAPI({
+      server: connectionString
+    });
+    this.address = address;
+    this.emitter = new EventEmitter();
+  }
+
+  async asyncCall(method, args) {
+    return new Promise((resolve) => {
+      this.rpc.connect().then(() => {
+        return this.rpc[method](args).then((result) => {
+          this.rpc.disconnect().then(() => {
+            return resolve(result);
+          });
+        });
+      });
+    });
+  }
+
+  async asyncRequest(method, args) {
+    return new Promise((resolve) => {
```

Comment:
Could we do dis?
`return this.asyncCall('request', args)`

Reply:
Fixed